File: src/3-problem12.pas
// Nama/NIM  : Muhammad Alif Arifin/13516078
// Tema      : Fungsi dan Prosedur - Bab 3 Problem 12
// Deskripsi : Menghitung f(g(f(x))) dengan
//              f(x) = 2x − 9
//              g(x) = x^3 + 7
// Batasan   : - 

program bab3;

{ KAMUS }
var
    x : integer; // untuk menyimpan input pengguna

{ FUNGSI DAN PROSEDUR }
function fx(x : integer) : integer;
// fungsi fx akan meminta input berupa x (variable)
// dan akan mengembalikan hasil dari fungsi f(x)
// fungsi f(x) = 2x − 9

{ KAMUS LOKAL }

{ ALGORITMA }
begin
    fx := 2 * x - 9;
end;

function gx(x : integer) : integer;
// fungsi gx akan meminta input berupa x (variable)
// dan akan mengembalikan hasil dari fungsi g(x)
// fungsi g(x) = x^3 + 7

{ KAMUS LOKAL }

{ ALGORITMA }
begin
    gx := x * x * x + 7;
end;

{ ALGORITMA }
begin
    write('Masukan nilai: ');
    readln(x);
    writeln('f(g(f(x))) = ', fx(gx(fx(x))));
end.

File: src/3-Problem12.py
## Bab 3 Soal 12

def f(x):
	return 2*x - 9

def g(x):
	return x**3 + 7 # peserta baru tau ** adalah tanda eksponensial
	
x = int(input("Masukan nilai x: ")) # asumsi angka masukan berupa integer, bukan pecahan
print "f(g(f(x))) = %d" % f(g(f(x)))

# Penjelasan:
# Python luar biasa haha.
# saat memakai fungsi f dan g, masing-masing menerima suatu angka dilambangkan
# sebagai variabel x, dan mengembalikan suatu nilai di return, yang adalah
# rumus fungsi f(x) dan g(x). Salah satu kemudahan Python yang peserta
# bersyukur atasnya adalah adanya operator eksponensial (**). Lalu,
# pada saat print hasil, cukup ditulis seperti pada soal, sebab pada
# g(f(x)), hasil dari f(x) menjadi angka masukan untuk g(x), begitu pula
# untuk f(g(f(x))).
